t/test_tgpolyt_depths.py

```diff
@@ -0,0 +1,130 @@
+
+    if not ('(TG)11-7T', 16) in tgpolyt_counts[0].items():
+        logger.error('could not find variant (TG)11-7T with 16 reads: {}'.format(tgpolyt_counts))
+        assert 0
+    pass
+
+def test_get_tgpolyt_bases():
+    '''
+    test get_tgpolyt_bases
+    :return:
+    '''
+    logger.info('testing get_tgpolyt_bases()')
+
+    bampysam = pysam.Samfile( bam_file, "rb")
+    read_dict = {x.query_name: x for x in bampysam.fetch()}
+
+    read = read_dict['HISEQ1:17:H947YADXX:1:2213:5880:62287']
+    assert get_tgpolyt_bases(read,REGION[1], REGION[2]) == 'ATGTGTGTGTGTGTGTGTGTGTGTTTTTTTA'
+
+    read = read_dict['HISEQ1:17:H947YADXX:2:1208:7897:88820']
+    assert get_tgpolyt_bases(read, REGION[1], REGION[2]) == ''
+
+    # sequence that possibly causes an issue when calling polyT
+    read = read_dict['HISEQ1:17:H947YADXX:1:2104:17965:28539']
+    assert get_tgpolyt_bases(read, REGION[1], REGION[2]) == 'ATGTGTGTGTGTTTGTGTGTGTGTTTTTTTA'
+
+    pass
+
+def test_count_tg_t_01():
+    '''
+    testing count_tg_t under different scenarios
+    '''
+    logger.info('testing count_tg_t()')
+    # easy sequence
+    variant = count_tg_t("ACTTTGTGTGTTTTTATGTG")
+    assert variant == "(TG)3-5T"
+
+    # ambigous sequence with (TG)5-2T and (TG)5-7T variants
+    variant = count_tg_t("ATGTGTGTGTGTTTGTGTGTGTGTTTTTTTA")
+    if variant != "(TG)5-7T":
+        logger.warning('missing variant (TG)5-7T {}'.format(variant))
+    assert variant == "(TG)5-2T"
+
+    # polyT sequence ending in TG which should yield (TG)3-5T
+    variant = count_tg_t("ACTTGTGTGTTTTTGTG")
+    if variant != "(TG)3-5T":
+        logger.warning('missing variant (TG)3-5T {}'.format(variant))
+    assert variant == "(TG)3-4T"
+
+    # precessing TG but not followed by polyT
+    variant = count_tg_t("ACTTGAATGTGTTTTTGTG")
+    if variant != "(TG)2-5T":
+        logger.warning('missing variant (TG)2-5T {}'.format(variant))
+    assert variant == "(TG)1-0T"
+
+
+    pass
+
+def test_hethom_call():
+    '''
+    testing different scenarios with hethom_call
+
+    '''
+    logger.info('testing hethom_call()')
+
+    assert hethom_call(['11-7T'], [float(16) / 17]) == 'hom'
+
+    assert hethom_call(['11-9T'], [float(16) / 17]) == 'hom alt'
+
+    assert hethom_call(['11-9T'], [float(8) / 20]) == 'het'
+
+    assert hethom_call(['11-9T'], [float(5) / 20]) == '--'
+
+    assert hethom_call(['11-7T', '5-2T'], [float(16)/17, float(1)/17]) == 'hom/--'
+
+    assert hethom_call(['11-9T', '5-2T'], [float(7) / 20, float(8) / 20]) == 'het/het'
+
+    assert hethom_call(['11-9T', '5-2T'], [float(4) / 20, float(4) / 20]) == '--/--'
+
+    assert hethom_call(['11-9T', '11-7T', '5-2T'], [float(10) / 30, float(10) / 30, float(10) / 30]) == 'het/het/het'
+
+    assert hethom_call(['11-9T', '5-2T'], [float(16) / 20, float(4) / 20], hom_freq_threshold = 0.75) == 'hom alt/--'
+
+    assert hethom_call(['11-9T', '5-2T'], [float(7) / 20, float(4) / 20], hom_freq_threshold =  0.35) == 'het/--'
+
+    pass
```

Comment:
if there is no newline at the end of the file, whenever you change the last line of the file it comes up as this line changing.

there's a setting in pycharm i added to make it always add a newline:
https://stackoverflow.com/questions/36043061/automatically-add-newline-on-save-in-pycharm

you also don't need a pass here